### PR TITLE
Add HealthStatus method to check health of service and revision

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
Add HealthStatus method to check health of service and revision of CloudRun.
I will explain each health condition below.

```
- If there are only True Status in condition section, model.CloudRunResourceState_HEALTHY is returned.
- If there is at least one Unknown or False Status in condition section, the one defined earlier is returned.
  For example, model.CloudRunResourceState_OTHER is returned if the first status in condition section is false.
```

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/cloudprovider/cloudrun/client.go [hunk 1]
```diff
@@ -185,3 +185,16 @@ func manifestToRunService(sm ServiceManifest) (*run.Service, error) {
 	}
 	return &s, nil
 }
+
+func manifestToRunRevision(rm RevisionManifest) (*run.Revision, error) {
```
Comment (khanhtc1202) [inline pkg/app/piped/cloudprovider/cloudrun/client.go:189]: It looks like we don't need this, isn't it?

File: pkg/app/piped/cloudprovider/cloudrun/cloudrun.go [hunk 2]
```diff
@@ -146,6 +148,30 @@ func (s *Service) ActiveRevisionNames() []string {
...
 }
 
+func (s *Service) HealthStatus() (model.CloudRunResourceState_HealthStatus, string) {
+	var (
+		status     = s.Status
+		errMessage = "Unexpected error while calculating: %s"
+	)
+	if status == nil {
+		msg := fmt.Sprintf(errMessage, "unable to find status")
+		return model.CloudRunResourceState_UNKNOWN, msg
```
Comment (khanhtc1202) [inline pkg/app/piped/cloudprovider/cloudrun/cloudrun.go:158]:
```suggestion
		return model.CloudRunResourceState_UNKNOWN, fmt.Sprintf(errMessage, "unable to find status")
```
nits, same with the below

Comment (knanao) [reply pkg/app/piped/cloudprovider/cloudrun/cloudrun.go]: Thanks for reviewing.
I fixed from reviews. PTAL when you can do.

File: pkg/app/piped/cloudprovider/cloudrun/client_test.go [hunk 3]
```diff
@@ -57,3 +59,26 @@ func TestManifestToRunService(t *testing.T) {
...
+	rm, err := ParseRevisionManifest([]byte(revisionManifest))
+	require.NoError(t, err)
+	require.NotEmpty(t, rm)
+
+	got, err := manifestToRunRevision(rm)
+	require.NoError(t, err)
+	assert.NotEmpty(t, got)
+}
+
+func manifestToRunRevision(rm RevisionManifest) (*run.Revision, error) {
```
Comment (khanhtc1202) [inline pkg/app/piped/cloudprovider/cloudrun/client_test.go:73]: sorry, finally I can understand what do you want here 🙏  I think we can make it as RevisonManifest method and move it here
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/cloudprovider/cloudrun/revisionmanifest.go
same with `manifestToRunService` used at L28, should move to `ServiceManifest`
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/cloudprovider/cloudrun/servicemanifest.go

Comment (knanao) [reply pkg/app/piped/cloudprovider/cloudrun/client_test.go:73]: Exactly, let me apply it.

**Comment by khanhtc1202** [comment]
Nice
/lgtm

**Comment by nghialv** [comment]
Great work!
/approve

**Pull request merged**